### PR TITLE
Restore `open-all-notifications` feature

### LINKED ISSUE #2911: Notifications Beta: `open-all-notifications` doesn't work
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/refined-github/issues/2911)

Not entirely sure if you are currently interested in fixing issues related to new notifications (still in beta I think?) but the [`open-all-notifications`](https://github.com/sindresorhus/refined-github/blob/master/source/features/open-all-notifications.tsx) doesn't work anymore - there's no "Open all in tabs" button anywhere.

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $10.00)

- [janpio janpio](https://issuehunt.io/u/janpio) ($10.00)

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/refined-github/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

### PR DESCRIPTION
**Pull request by Jolg42**
1. LINKED ISSUES:
   Closes #2911

2. TEST URLS:
   https://github.com/notifications

3. SCREENSHOT:
Screen Shot 2020-05-01 at 16 06 35

I fixed the "Open all unread in tabs" for the new notification page that is now per default.
I removed the `addOpenReposButton` as it's broken anyway.

It will make @janpio really happy 😉

### CODE REVIEW

File: source/features/open-all-notifications.css [hunk 1]
```diff
@@ -1,15 +0,0 @@
...
-	padding: 3px 10px;
-	border: none;
-	margin-left: 2px;
-	color: #586069;
-	background: none;
-}
-
-.mark-all-as-read-confirmed .open-repo-notifications {
-	visibility: hidden;
-}
```
Comment (fregante) [inline source/features/open-all-notifications.css:15]: 🙌

File: source/features/open-all-notifications.tsx [hunk 2]
```diff
@@ -27,55 +24,24 @@ function openNotifications({delegateTarget}: delegate.Event): void {
...
-		const button = <button className="btn btn-sm rgh-open-notifications-button" type="button">Open all unread in tabs</button>;
-		select('.tabnav .float-right')!.prepend(button);
-
-		// There is no sibling on `/<org>/<repo>/notifications` page
-		if (button.nextElementSibling) {
-			groupButtons([button, button.nextElementSibling]);
-		}
+		// Create an open button and add it to the DOM
+		const button = <button className="btn btn-sm ml-2 rgh-open-notifications-button" type="button">Open all unread in tabs</button>;
+		select('.notifications-list .Box-header')!.append(button);
```
Comment (fregante) [inline source/features/open-all-notifications.tsx:44]: This fails when grouping by repository:

Screen Shot 2020-05-01 at 16 32 01

Comment (fregante) [reply source/features/open-all-notifications.tsx:44]: Also I'd keep the icon there and I'd skip the `button` variable. Example:

```ts
const button = ;
select('.notifications-list .Box-header')!.append(
	<button className="btn btn-sm ml-2 rgh-open-notifications-button" type="button">
		<LinkExternalIcon/> Open all unread in tabs
	</button>
);
```

**Comment by fregante** [comment]
Once thing to notice: the old Notifications were grouped by Repository (option still available today, but not as default) and this function had a "Open all unread" button _on each repository._



This PR removed this specific feature. I think it was somewhat useful, what do you say?

**Comment by Jolg42** [comment]
New location 🥇 

It works in both views, complete or per repo:

Screen Shot 2020-05-01 at 17 26 02


Screen Shot 2020-05-01 at 17 26 11

File: source/features/open-all-notifications.tsx (hunk 2 above)
Comment (Jolg42) [reply source/features/open-all-notifications.tsx]: Location updated see comments 😉

**Comment by Jolg42** [comment]
https://github.com/sindresorhus/refined-github/pull/3046#issuecomment-622418616

@fregante How do I access this view 🙃

**Comment by fregante** [comment]
> How do I access this "old" view? 🙃 I don't think it's possible to see it anymore (or I don't see how)

Can't, that was an old gif

**Comment by fregante** [comment]
> New location 🥇

Not sure I like it, it's too big. I'd rather make the feature look more like `batch-option-issues` than more different, especially since Notifications now look a lot like issue/PR lists.

**Comment by Jolg42** [comment]
@fregante I have no idea what is `batch-option-issues` please be more specific ...

**Comment by Jolg42** [comment]
Like this @fregante?
Screen Shot 2020-05-01 at 18 39 47

**Comment by netniV** [comment]
I used to open issues per repo at a time so I knew the only issues I was looking at pertained to a particular problem, and I have many repos I review so just opening ALL repos is a bad suggestion for me.  This was especially useful for going down the notification list and opening per repo without needing to move from/on it.

Can't tell from the comments above whether that would exist in the beta version currently?

**Comment by Jolg42** [comment]
@netniV the "Open all repos" was broken since the notifications beta is now a default so I removed it.

**Comment by fregante** [comment]
> Like this @fregante?

You had it right at the very beginning. I was suggesting **adding** it to each grouped repo only if they're grouped by repo.

**Comment by fregante** [comment]
> @fregante I have no idea what is `batch-option-issues` please be more specific.

Sorry, I meant [`batch-open-issues`](https://github.com/sindresorhus/refined-github/blob/master/source/features/batch-open-issues.tsx)

Screen Shot 2020-05-01 at 19 53 41

**Comment by fregante** [comment]
I think you misunderstood what I mean by "Grouped by repo". There's a button next to the search field: 
Screen Shot 2020-05-01 at 16 32 01

File: source/features/open-all-notifications.tsx (hunk 2 above)
Comment (fregante) [reply source/features/open-all-notifications.tsx]: Neither one of my suggestion was taken. Please resolve

Comment (Jolg42) [reply source/features/open-all-notifications.tsx]: Grouping by repository works now

**Comment by Jolg42** [comment]
@fregante 

Normal view:
Screen Shot 2020-05-01 at 22 44 00

Grouped by repository:
Screen Shot 2020-05-01 at 22 45 09

I think this is ready to be merged 😃 

About 
```
// Hide button when the contextual actions are visible
.js-notifications-mark-selected-actions:not([hidden]) .rgh-open-notifications-button {
	display: none;
}
```
If you find a way to make it work let me know, I think it's impossible without JavaScript and I don't think it's a good idea to do it in JS.

Also I agree the icon alone is nice in the group by repo view but for the default view (by date) having one button that says "Open all unread" is better, easier to find & understand on first sight!

**Comment by Jolg42** [comment]
I tested different scenario and everything works so far

**Comment by fregante** [comment]
> I think this is ready to be merged 😃

Please look again at my screenshot, the position and style of the buttons don't match it. Match it entirely before requesting another review. 

It's possible that reverting the whole PR, adjusting the selectors and classes will make it easier to make it work like before.

**Comment by fregante** [comment]
> If you find a way to make it work let me know, I think it's impossible without JavaScript

It's possible without JS with the CSS snippet I posted earlier, except I forgot to add the `~`:

```css
.js-notifications-mark-selected-actions:not([hidden]) ~ .rgh-open-notifications-button {
	display: none;
}
```

Here's a gif based on 99d25e5c + the CSS snippet above.

demo2

**Comment by Jolg42** [comment]
@fregante I moved the html element though and put it fixed on the right side.

I find it easier to have it there / it doesn't move and it's more in line with the grouped repository view. 

But it means that you're snippet doesn't works anymore in the latest version of the code.

I don't think it's a blocker 🤷‍♂️

**Comment by Jolg42** [comment]
Also for the style, it was impossible to know what you wanted ...

Right now it's like the open issue like you wanted but now you're telling me it's not what you wanted, it's confusing for both of us.
Please try to make specific comments when possible 😃

**Comment by fregante** [comment]
> Also for the style, it was impossible to know what you wanted ...

I'm sorry what? I posted a screenshot, I literally could not be any clearer.

I repeat:
- you're not using the `btn` class anywhere
- the repo buttons aren't grouped
- the repo buttons lack a label
- the main button does not disappear when the user select some (the selector still works, except something like `.rgh-open-notifications-button` needs to be added to the `.d-flex` element you added)

Post again when they're done

**Pull request closed without merging**